src/index.js

```diff
@@ -60,77 +64,80 @@ export default class AutoCheckElement extends HTMLElement {
   }
 
   set required(required: boolean) {
-    this.input.required = required
+    const input = this.input
+    if (!input) return
+
+    input.required = required
     if (required) {
       this.setAttribute('required', '')
     } else {
       this.removeAttribute('required')
     }
   }
+}
 
-  check() {
-    if (!this.src) {
-      throw new Error('missing src')
-    }
-    if (!this.csrf) {
-      throw new Error('missing csrf')
-    }
+function check(autoCheckElement: AutoCheckElement) {
+  if (!autoCheckElement.src) {
+    throw new Error('missing src')
+  }
+  if (!autoCheckElement.csrf) {
+    throw new Error('missing csrf')
+  }
+  const input = autoCheckElement.input
+  if (!input) return
 
-    const body = new FormData()
-    body.append('authenticity_token', this.csrf)
-    body.append('value', this.input.value)
+  const body = new FormData()
+  body.append('authenticity_token', autoCheckElement.csrf)
```

Comment:
Since `.csrf` is computed programatically via getter, could we save it to a const within this method so we don't access it more than once? Same for `.src`, which is even more expensive since it creates an element